smurfs/src/actions/index.js

```diff
@@ -1,15 +1,74 @@
-/* 
-  Action Types Go Here!
-  Be sure to export each action type so you can pull it into your reducer
-*/
-
-/*
-  For this project you'll need at least 2 action creators for the main portion,
-   and 2 more for the stretch problem.
-   Be sure to include action types for each type of action creator. Also, be sure to mind
-     the "pending" states like, fetching, creating, updating and deleting.
-   C - addSmurf
-   R - getSmurfs
-   U - updateSmurf
-   D - deleteSmurf
-*/
+// --------------------------- DEPENDANCIES -------------------------------------------//
+import axios from 'axios';
+// --------------------------- DEPENDANCIES -------------------------------------------//
+
+// --------------------------- ACTION EXPORTS -------------------------------------------//
+export const FETCH_SMURFS = 'FETCH_SMURFS';
+export const FETCHED_SMURFS = 'FETCHED_SMURFS';
+export const ADDING_SMURF = 'ADDING_SMURF';
+export const ADDED_SMURF = 'ADD_SMURF';
+export const DELETING_USER = 'DELETING_SMURF';
+export const DELETED_USER = 'DELETED_SMURF';
+export const UPDATING_SMURF = 'UPDATING_SMURF';
+export const UPDATED_SMURF = 'UPDATED_SMURF';
+export const ERROR = 'ERROR';
+// --------------------------- ACTION EXPORTS -------------------------------------------//
+
+// --------------------------- DISPATCHES -------------------------------------------//
+export const getSmurfs = () => {
+  return dispatch => {
+    dispatch({ type: FETCH_SMURFS });
+    axios
+      .get(`http://localhost:3333/smurfs`)
+      .then(res => {
+        dispatch({ type: FETCHED_SMURFS, payload: res.data });
+      })
+      .catch(err => dispatch({ type: ERROR, error: err }));
+  };
+};
+
+export const addSmurf = smurf => {
```

Comment:
I think you are totally ready to use destructuring ... you don't have to technically but I think you will like the time it saves, and the readability it conveys:
```suggestion
export const addSmurf = ( { smurf } ) => {